shop.rb

```diff
@@ -32,23 +33,31 @@ class App < Sinatra::Base
     get "/products/:id" do |id|
       product = FindProduct.new.call(id)
       halt 404 unless product
+      erb :"products/show", locals: { product: product }
     end
 
     get "/warehouse" do
-      products = FetchWarehouseItems.new.call
+      items = FetchWarehouseItems.new.call
+      erb :"warehouse/index", locals: { items: items, title: "Warehouse" }
```

Comment:
I was trying to make here something analogous to "get" which refers to "/basket" but nothing happens; i don't know if it's only the matter of that the action isn't visible, or this service doesn't work. Should I put some another service which refers to warehouse in line 53, in post "/basket" method?